mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -243,15 +243,21 @@ public static boolean hiveCatalog(Configuration conf, Properties props) {
    */
   private static Map<String, String> addCatalogPropertiesIfMissing(Configuration conf, String catalogType,
                                                                    Map<String, String> catalogProperties) {
-    if (catalogType != null) {
-      catalogProperties.putIfAbsent(CatalogUtil.ICEBERG_CATALOG_TYPE, catalogType);
-    }
-
     String legacyCatalogImpl = conf.get(InputFormatConfig.CATALOG_LOADER_CLASS);
     if (legacyCatalogImpl != null) {
       catalogProperties.putIfAbsent(CatalogProperties.CATALOG_IMPL, legacyCatalogImpl);
     }
 
+    if (catalogProperties.containsKey(CatalogUtil.ICEBERG_CATALOG_TYPE) &&
+            catalogProperties.containsKey(CatalogProperties.CATALOG_IMPL)) {
```

Comment:
Indentation isn't correct in this file. Continuation indents should be 4 spaces.